### PR TITLE
fix redis jemalloc compatibility

### PR DESCRIPTION
**Pull request by blueSherii**

### CODE REVIEW

File: dev-db/redis/redis-7.4.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+"
+
+REQUIRED_USE="?? ( jemalloc tcmalloc )"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-6.2.1-config.patch
+	"${FILESDIR}"/${PN}-7.2.0-system-jemalloc.patch
+	"${FILESDIR}"/${PN}-6.2.3-ppc-atomic.patch
+	"${FILESDIR}"/${PN}-sentinel-7.2.0-config.patch
+	"${FILESDIR}"/${PN}-7.0.4-no-which.patch
```
Comment (zegeek) [inline dev-db/redis/redis-7.4.0.ebuild:54]: these patch files are missing from the filesdir, so the patching phase might fail, we have to retrieve them too from gentoo's git history

File: dev-db/redis/files/redis-5.0-shared.patch [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+@@ -120,9 +113,8 @@ ifeq ($(MALLOC),tcmalloc_minimal)
+ endif
+ 
+ ifeq ($(MALLOC),jemalloc)
+-	DEPENDENCY_TARGETS+= jemalloc
+-	FINAL_CFLAGS+= -DUSE_JEMALLOC -I../deps/jemalloc/include
+-	FINAL_LIBS := ../deps/jemalloc/lib/libjemalloc.a $(FINAL_LIBS)
++	FINAL_CFLAGS+= -DUSE_JEMALLOC -DJEMALLOC_NO_DEMANGLE
++	FINAL_LIBS+= -ljemalloc -ldl
+ endif
```
Comment (zegeek) [inline dev-db/redis/files/redis-5.0-shared.patch:30]: we could remove all this block from the patch file, or just remove the invocation of the patch file in the ebuilds themselves

**Review by zegeek — DISMISSED** [review]

File: dev-db/redis/redis-6.2.14-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+		dev-lang/tcl:0=
+		ssl? ( dev-tcltk/tls )
+	)"
+
+REQUIRED_USE="?? ( jemalloc tcmalloc )
+	${LUA_REQUIRED_USE}"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-6.2.1-config.patch
+	"${FILESDIR}"/${PN}-5.0-shared.patch
```
[the remark this answers is not in the document]
Comment (blueSherii) [reply dev-db/redis/redis-6.2.14-r1.ebuild]: `5.0-shared` was completely removed from the ebuilds

File: dev-db/redis/redis-7.0.15-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+	test? (
+		dev-lang/tcl:0=
+		ssl? ( dev-tcltk/tls )
+	)"
+
+REQUIRED_USE="?? ( jemalloc tcmalloc )"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-6.2.1-config.patch
+	"${FILESDIR}"/${PN}-5.0-shared.patch
```
[the remark this answers is not in the document]
Comment (blueSherii) [reply dev-db/redis/redis-7.0.15-r1.ebuild]: `5.0-shared` was completely removed from the ebuilds

**Pull request merged**